### PR TITLE
loaddata: revert concurrent writing to TiKV and WITH detached

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/44078

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/realtikvtest/loaddatatest/detach_test.go [hunk 1]
```diff
@@ -1,108 +0,0 @@
-// Copyright 2023 PingCAP, Inc.
```
Comment (D3Hunter) [inline tests/realtikvtest/loaddatatest/detach_test.go:1]: can keep this, will change it in IMPORT INTO

File: tests/realtikvtest/loaddatatest/one_parquet_test.go [hunk 2]
```diff
@@ -1,74 +0,0 @@
...
-	"fmt"
-	"time"
-
-	"github.com/fsouza/fake-gcs-server/fakestorage"
-	"github.com/pingcap/tidb/executor/importer"
-	"github.com/pingcap/tidb/testkit"
-	"github.com/stretchr/testify/require"
-)
-
-//go:embed test.parquet
```
Comment (D3Hunter) [inline tests/realtikvtest/loaddatatest/one_parquet_test.go:28]: this too

File: tests/realtikvtest/loaddatatest/test.parquet
Comment (D3Hunter) [inline tests/realtikvtest/loaddatatest/test.parquet:1]: diitto

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Review by D3Hunter — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**